python/lsst/meas/modelfit/priors/priors.cc

```diff
@@ -0,0 +1,137 @@
+    cls.def_static("getUpdateRestriction", &Class::getUpdateRestriction,
+                   py::return_value_policy::reference); // returned object has static duration
+    cls.def("getMixture", &Class::getMixture);
+    // virtual methods already wrapped by Prior base class
+}
+
+void declareSemiEmpirical(py::module & mod) {
+    using Class = SemiEmpiricalPrior;
+    using Control = SemiEmpiricalPriorControl;
+    using PyControl = py::class_<Control, std::shared_ptr<Control>>;
+    using PyClass = py::class_<Class, std::shared_ptr<Class>, Prior>;
+
+    PyControl clsControl(mod, "SemiEmpiricalPriorControl");
+    clsControl.def(py::init<>());
+    LSST_DECLARE_CONTROL_FIELD(clsControl, Control, ellipticitySigma);
+    LSST_DECLARE_CONTROL_FIELD(clsControl, Control, ellipticityCore);
+    LSST_DECLARE_CONTROL_FIELD(clsControl, Control, logRadiusMinOuter);
+    LSST_DECLARE_CONTROL_FIELD(clsControl, Control, logRadiusMinInner);
+    LSST_DECLARE_CONTROL_FIELD(clsControl, Control, logRadiusMu);
+    LSST_DECLARE_CONTROL_FIELD(clsControl, Control, logRadiusSigma);
+    LSST_DECLARE_CONTROL_FIELD(clsControl, Control, logRadiusNu);
+    clsControl.def("validate", &Control::validate);
+
+    PyClass cls(mod, "SemiEmpiricalPrior");
+    cls.def(py::init<Control>(), "ctrl"_a);
+    cls.attr("Control") = clsControl;
+    // virtual methods already wrapped by Prior base class
+}
+
+void declareSoftenedLinear(py::module & mod) {
+    using Class = SoftenedLinearPrior;
+    using Control = SoftenedLinearPriorControl;
+    using PyControl = py::class_<Control, std::shared_ptr<Control>>;
+    using PyClass = py::class_<Class, std::shared_ptr<Class>, Prior>;
+
+    PyControl clsControl(mod, "SoftenedLinearPriorControl");
+    clsControl.def(py::init<>());
+    LSST_DECLARE_CONTROL_FIELD(clsControl, Control, ellipticityMaxOuter);
+    LSST_DECLARE_CONTROL_FIELD(clsControl, Control, ellipticityMaxInner);
+    LSST_DECLARE_CONTROL_FIELD(clsControl, Control, logRadiusMinOuter);
+    LSST_DECLARE_CONTROL_FIELD(clsControl, Control, logRadiusMinInner);
+    LSST_DECLARE_CONTROL_FIELD(clsControl, Control, logRadiusMaxOuter);
+    LSST_DECLARE_CONTROL_FIELD(clsControl, Control, logRadiusMaxInner);
+    LSST_DECLARE_CONTROL_FIELD(clsControl, Control, logRadiusMinMaxRatio);
+
+    PyClass cls(mod, "SoftenedLinearPrior");
+    cls.def(py::init<Control>(), "ctrl"_a);
+    cls.def("getControl", &Class::getControl, py::return_value_policy::copy);
+    cls.attr("Control") = clsControl;
+    // virtual methods already wrapped by Prior base class
+}
+
+
+PYBIND11_PLUGIN(priors) {
+
+    py::module mod("priors");
+    py::module::import("lsst.meas.modelfit.mixture");
+
+    if (_import_array() < 0) {
+        PyErr_SetString(PyExc_ImportError, "numpy.core.multiarray failed to import");
+        return nullptr;
+    }
+
+    declarePrior(mod);
+    declareMixture(mod);
+    declareSemiEmpirical(mod);
+    declareSoftenedLinear(mod);
```

Comment:
I would suffix these with `Prior` to match the class names, but not strictly required by our rules.
Also, while I think mixing these in one wrapper file is not actually needed here it also doesn't seem so bad. But others may have a different opinion.